packages/notebook/src/actions.tsx

```diff
@@ -1314,6 +1316,81 @@ export namespace NotebookActions {
     Private.handleState(notebook, state, true);
   }
 
+  /**
+   * Render side-by-side.
+   *
+   * @param panel - The current notebook panel.
+   */
+  export function renderSideBySide(
+    panel: NotebookPanel,
+    app?: JupyterLab
+  ): void {
+    Private.sideBySide = true;
+    const applySideBySide = () => {
+      if (!Private.sideBySide) {
+        return;
+      }
+
+      const halfWidth = panel.node.clientWidth / 2;
+      const nodes = document.getElementsByClassName('jp-CodeCell');
+      for (let i = 0; i < nodes.length; i++) {
+        const ele = nodes.item(i) as HTMLElement;
+        ele.style.display = 'flex';
+        ele.style.alignItems = 'flex-start';
+        ele.style.flexWrap = 'wrap';
+        ele.style.direction = 'row';
+      }
+      const inputarea = document.getElementsByClassName('jp-InputArea-editor');
+      for (let j = 0; j < inputarea.length; j++) {
+        const area = inputarea.item(j) as HTMLElement;
+        area.style.minWidth = `calc(${halfWidth}px - 30px)`;
+        area.style.maxWidth = `calc(${halfWidth}px - 30px)`;
```

Comment:
Do not emulate `grid` with `flex` with manual calculations; this will lead to overflow issues. Use CSS grid instead.